### PR TITLE
Add more structure to the console logs

### PR DESCRIPTION
**Pull request by vladaionescu**
Example output: https://github.com/earthly/earthly/pull/1226/checks?check_run_id=3726613288#step:11:1

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -444,17 +451,28 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
...
+			err = b.s.buildMainMulti(ctx, bf, onImage, onArtifact, onFinalArtifact, onPull, PHASE_PUSH)
 			if err != nil {
 				return nil, errors.Wrapf(err, "build push")
 			}
 		}
 		sp.printCurrentSuccess()
+	} else {
+		if opt.PrintPhases {
+			b.opt.Console.PrintPhaseHeader(PHASE_PUSH, true, "")
+			b.opt.Console.Printf("To enable pushing use\n\n\t\tearthly --push ...\n")
```
Comment (alexcb) [inline builder/builder.go:463]: nice addition!

**Review by alexcb — APPROVED** [review]

File: conslogging/delayed.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	dl.queue = append(dl.queue, fmt.Sprintf(format, v...))
+}
+
+// Flush prints the queued up messages to the underlying console.
+func (dl *DelayedLogger) Flush() {
+	for _, s := range dl.queue {
+		dl.cl.Printf("%s", s)
+	}
+	dl.queue = nil
+}
```
Comment (vladaionescu) [inline conslogging/delayed.go:35]: Flush was written entirely by GH Copilot if you can believe that.

File: conslogging/delayed.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package conslogging
+
+import (
+	"fmt"
+)
+
+// DelayedLogger is a logger that queues up messages until Flush is called.
+type DelayedLogger struct {
```
Comment (alexcb) [inline conslogging/delayed.go:8]: nit: usually writers that have a `Flush` method are called buffered. e.g. go's `bufio`.

```suggestion
type BufferedLogger struct {
```

**Review by alexcb — APPROVED** [review]

**Comment by vladaionescu** [comment]
I'll merge this after the release.

**Pull request merged**